Sources/NIOCore/EventLoopFuture+AssumeIsolated.swift

```diff
@@ -456,6 +456,29 @@ extension EventLoopFuture {
             }
         }
 
+        /// Wait for the resolution of this `EventLoopFuture` by blocking the current thread until it
+        /// resolves.
+        ///
+        /// If the `EventLoopFuture` resolves with a value, that value is returned from `wait()`. If
+        /// the `EventLoopFuture` resolves with an error, that error will be thrown instead.
+        /// `wait()` will block whatever thread it is called on, so it must not be called on event loop
+        /// threads: it is primarily useful for testing, or for building interfaces between blocking
+        /// and non-blocking code.
+        ///
+        /// This is also forbidden in async contexts: prefer ``EventLoopFuture/get()``.
+        ///
+        /// - Note: The `Value` must be `Sendable` since it is shared outside of the isolation domain of the event loop.
+        ///
+        /// - returns: The value of the `EventLoopFuture` when it completes.
+        /// - throws: The error value of the `EventLoopFuture` if it errors.
+        @available(*, noasync, message: "wait() can block indefinitely, prefer get()", renamed: "get()")
+        @preconcurrency
+        @inlinable
+        public func wait(file: StaticString = #file, line: UInt = #line) throws -> Value {
```

Comment:
Also, this method is a bit of a code smell to me. In almost all cases this is a deterministic crash, because the only place you are _not_ allowed to call `wait` from is the event loop to which the future belongs. The only time it's safe is when you're on `EmbeddedEventLoop`.

A better fix moves this to a helper method on `EmbeddedEventLoop`.